config/v1/types_infrastructure.go

```diff
@@ -665,10 +665,8 @@ type AlibabaCloudPlatformStatus struct {
 	// +required
 	Region string `json:"region"`
 	// resourceGroupID is the ID of the resource group for the cluster.
-	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Pattern=`^rg-[0-9A-Za-z]+$`
-	// +required
-	ResourceGroupID string `json:"resourceGroupID"`
+	// +optional
+	ResourceGroupID string `json:"resourceGroupID,omitempty"`
```

Comment:
omitempty and required doesn't make sense. Why do you remove the kubebuilder pattern tag? It must stay.

Reply:
Moving from required to optional break downgrades. Is that considered?

Reply:
> Moving from required to optional break downgrades. Is that considered?

this is the first release of the api